README.md

```diff
@@ -893,6 +895,72 @@ export function buildFluidImageData(
 };
 ```
 
+### Third-party integration helpers (e.g. for Gatsby source plugins)
+
+This section is for third-party plugin developers of Gatsby plugins and sources that would like to expose the functionality of this library inside their own plugins natively. For example, the [Prismic Gatsby plugin](https://github.com/angeloashmore/gatsby-source-prismic) uses these APIs to expose `fixed`, `fluid`, `url`, and `gatsbyImageData` fields on its images that it exposes in its GraphQL API.
+
+#### createImgixGatsbyTypes
+
+This is the main function that should be used to reuse functionality from this library.
+
+##### API
+
+| Parameter          | Type                              | Required | Description                                                                                                                                                                                                                                                                       |
+| ------------------ | --------------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| imgixClientOptions | `Object`                          |          | Any extra configuration to pass to new ImgixClient from @imgix/js-core (see [here](https://github.com/imgix/js-core#configuration) for more information).                                                                                                                         |
+| resolveUrl         | `(source) => string \| undefined` | ✔        | A callback to resolve the image URL from the source data provided to the resolver. The source data is provided by the resolver of the type you are adding the image fields too.                                                                                                   |
+| resolveWidth       | `(source) => number \| undefined` |          | This callback can be used to provide the source (uncropped) width of the target image from the source data, if known. The data passed to this callback is the same as is passed to `resolveUrl`.                                                                                  |
+| resolveHeight      | `(source) => number \| undefined` |          | This callback can be used to provide the source (uncropped) height of the target image from the source data, if known. The data passed to this callback is the same as is passed to `resolveUrl`.                                                                                 |
```

Comment:
A quick note that could help is if the width/height are not known or provided, the plugin will automatically make a network request to fetch this information. Documenting this lets plugin authors know that it's not required to get this data if they don't have it, but also that if they don't have it, an extra network request per image will be made.